### ISSUE TITLE
Node shutdown process freezing for over 30 minutes during shut down database stage

### ISSUE THREAD

**Issue by ShlomiBalalis** [scylladb/scylladb#9242]
*Installation details*
Scylla version (or git commit hash): 4.5.rc6-0.20210817.88314fedf with build-id f1e63bebd85e29490480de9c25887e6078fa63f3
Cluster size: 5
OS (RHEL/CentOS/Ubuntu/AWS AMI): ami-011f02dc29a254fd4 (eu-west-1)
instance type: i3en.3xlarge
Scenario: [longevity-large-partition-200k-pks-4days-test](https://github.com/scylladb/scylla-cluster-tests/blob/branch-4.5/test-cases/longevity/longevity-large-partition-200k_pks-4days.yaml)

At `2021-08-24 04:43:32.931`, a RestartNodeWithResharding nemesis has started on node no. 10. The nemesis shuts down the node, set `murmur3_partitioner_ignore_msb_bits=15` and starts the node agein, while making sure that the resharding process ended successfully. Afterwards, the nemesis executes the process again and sets `murmur3_partitioner_ignore_msb_bits` back to the default (12).
In this run, however, when the nemesis tried to shut down the node during the second iteration (before setting `murmur3_partitioner_ignore_msb_bits` back to 12) the node's shutdown process "froze" for over 30 minutes, during which shard 0 was aborted:
```
2021-08-24T05:16:53+00:00  longevity-large-partitions-200k-pks-db-node-384d3370-10 !INFO    | scylla:  [shard 0] compaction_manager - Asked to stop
2021-08-24T05:16:53+00:00  longevity-large-partitions-200k-pks-db-node-384d3370-10 !INFO    | scylla:  [shard 0] compaction_manager - Stopping 1 ongoing compactions
2021-08-24T05:16:53+00:00  longevity-large-partitions-200k-pks-db-node-384d3370-10 !INFO    | systemd[1]: Stopping Scylla Server...
2021-08-24T05:16:53+00:00  longevity-large-partitions-200k-pks-db-node-384d3370-10 !INFO    | scylla:  [shard 0] init - Signal received; shutting down
2021-08-24T05:16:53+00:00  longevity-large-partitions-200k-pks-db-node-384d3370-10 !INFO    | scylla:  [shard 0] init - Shutting down redis service
2021-08-24T05:16:53+00:00  longevity-large-partitions-200k-pks-db-node-384d3370-10 !INFO    | scylla:  [shard 0] init - Shutting down redis service was successful
2021-08-24T05:16:53+00:00  longevity-large-partitions-200k-pks-db-node-384d3370-10 !INFO    | scylla:  [shard 0] init - Shutting down view builder
2021-08-24T05:16:53+00:00  longevity-large-partitions-200k-pks-db-node-384d3370-10 !INFO    | scylla:  [shard 0] view - Stopping view builder
2021-08-24T05:16:53+00:00  longevity-large-partitions-200k-pks-db-node-384d3370-10 !INFO    | scylla:  [shard 1] compaction_manager - Asked to stop
2021-08-24T05:16:53+00:00  longevity-large-partitions-200k-pks-db-node-384d3370-10 !INFO    | scylla:  [shard 2] compaction_manager - Asked to stop
2021-08-24T05:16:53+00:00  longevity-large-partitions-200k-pks-db-node-384d3370-10 !INFO    | scylla:  [shard 1] compaction_manager - Stopping 1 ongoing compactions
2021-08-24T05:16:53+00:00  longevity-large-partitions-200k-pks-db-node-384d3370-10 !INFO    | scylla:  [shard 10] compaction_manager - Asked to stop
2021-08-24T05:16:53+00:00  longevity-large-partitions-200k-pks-db-node-384d3370-10 !INFO    | scylla:  [shard 2] compaction_manager - Stopping 1 ongoing compactions
2021-08-24T05:16:53+00:00  longevity-large-partitions-200k-pks-db-node-384d3370-10 !INFO    | scylla:  [shard 1] view - Stopping view builder
2021-08-24T05:16:53+00:00  longevity-large-partitions-200k-pks-db-node-384d3370-10 !INFO    | scylla:  [shard 10] compaction_manager - Stopping 1 ongoing compactions
2021-08-24T05:16:53+00:00  longevity-large-partitions-200k-pks-db-node-384d3370-10 !INFO    | scylla:  [shard 10] view - Stopping view builder
2021-08-24T05:16:53+00:00  longevity-large-partitions-200k-pks-db-node-384d3370-10 !INFO    | scylla:  [shard 4] compaction_manager - Asked to stop
2021-08-24T05:16:53+00:00  longevity-large-partitions-200k-pks-db-node-384d3370-10 !INFO    | scylla:  [shard 6] compaction_manager - Asked to stop
2021-08-24T05:16:53+00:00  longevity-large-partitions-200k-pks-db-node-384d3370-10 !INFO    | scylla:  [shard 4] compaction_manager - Stopping 1 ongoing compactions
...
2021-08-24T05:17:00+00:00  longevity-large-partitions-200k-pks-db-node-384d3370-10 !INFO    | scylla:  [shard 0] init - Shutting down drain storage proxy was successful
2021-08-24T05:17:00+00:00  longevity-large-partitions-200k-pks-db-node-384d3370-10 !INFO    | scylla:  [shard 0] init - Shutting down streaming service
2021-08-24T05:17:00+00:00  longevity-large-partitions-200k-pks-db-node-384d3370-10 !INFO    | scylla:  [shard 0] init - Shutting down streaming service was successful
2021-08-24T05:17:00+00:00  longevity-large-partitions-200k-pks-db-node-384d3370-10 !INFO    | scylla:  [shard 0] init - Shutting down Raft services
2021-08-24T05:17:00+00:00  longevity-large-partitions-200k-pks-db-node-384d3370-10 !INFO    | scylla:  [shard 0] init - Shutting down Raft services was successful
2021-08-24T05:17:00+00:00  longevity-large-partitions-200k-pks-db-node-384d3370-10 !INFO    | scylla:  [shard 0] init - Shutting down storage proxy RPC verbs
2021-08-24T05:17:00+00:00  longevity-large-partitions-200k-pks-db-node-384d3370-10 !INFO    | scylla:  [shard 0] init - Shutting down storage proxy RPC verbs was successful
2021-08-24T05:17:00+00:00  longevity-large-partitions-200k-pks-db-node-384d3370-10 !INFO    | scylla:  [shard 0] init - Shutting down sstables format selector
2021-08-24T05:17:00+00:00  longevity-large-partitions-200k-pks-db-node-384d3370-10 !INFO    | scylla:  [shard 0] init - Shutting down sstables format selector was successful
2021-08-24T05:17:00+00:00  longevity-large-partitions-200k-pks-db-node-384d3370-10 !INFO    | scylla:  [shard 0] init - Shutting down migration manager
2021-08-24T05:17:00+00:00  longevity-large-partitions-200k-pks-db-node-384d3370-10 !INFO    | scylla:  [shard 0] init - Shutting down migration manager was successful
2021-08-24T05:17:00+00:00  longevity-large-partitions-200k-pks-db-node-384d3370-10 !INFO    | scylla:  [shard 0] init - Shutting down database
...
2021-08-24T05:31:53+00:00  longevity-large-partitions-200k-pks-db-node-384d3370-10 !WARNING | systemd[1]: scylla-server.service: State 'stop-sigterm' timed out. Killing.
2021-08-24T05:31:53+00:00  longevity-large-partitions-200k-pks-db-node-384d3370-10 !NOTICE  | systemd[1]: scylla-server.service: Killing process 787427 (scylla) with signal SIGABRT.
2021-08-24T05:31:53+00:00  longevity-large-partitions-200k-pks-db-node-384d3370-10 !INFO    | scylla[787427]: Aborting on shard 0.
2021-08-24T05:31:53+00:00  longevity-large-partitions-200k-pks-db-node-384d3370-10 !INFO    | scylla[787427]: Backtrace:
2021-08-24T05:31:53+00:00  longevity-large-partitions-200k-pks-db-node-384d3370-10 !INFO    | scylla[787427]:   0x3f6fb58
2021-08-24T05:31:53+00:00  longevity-large-partitions-200k-pks-db-node-384d3370-10 !INFO    | scylla[787427]:   0x3fa1ca2
2021-08-24T05:31:53+00:00  longevity-large-partitions-200k-pks-db-node-384d3370-10 !INFO    | scylla[787427]:   0x7fa5b79ea1df
2021-08-24T05:31:53+00:00  longevity-large-partitions-200k-pks-db-node-384d3370-10 !INFO    | scylla[787427]:   /opt/scylladb/libreloc/libc.so.6+0xfc30c
2021-08-24T05:31:53+00:00  longevity-large-partitions-200k-pks-db-node-384d3370-10 !INFO    | scylla[787427]:   0x3fc9620
2021-08-24T05:31:53+00:00  longevity-large-partitions-200k-pks-db-node-384d3370-10 !INFO    | scylla[787427]:   0x3fc6074
2021-08-24T05:31:53+00:00  longevity-large-partitions-200k-pks-db-node-384d3370-10 !INFO    | scylla[787427]:   0x3fc6773
2021-08-24T05:31:53+00:00  longevity-large-partitions-200k-pks-db-node-384d3370-10 !INFO    | scylla[787427]:   0x3f83e9c
2021-08-24T05:31:53+00:00  longevity-large-partitions-200k-pks-db-node-384d3370-10 !INFO    | scylla[787427]:   0x3f21905
2021-08-24T05:31:53+00:00  longevity-large-partitions-200k-pks-db-node-384d3370-10 !INFO    | scylla[787427]:   0x3f20d26
2021-08-24T05:31:53+00:00  longevity-large-partitions-200k-pks-db-node-384d3370-10 !INFO    | scylla[787427]:   0xe8c184
2021-08-24T05:31:53+00:00  longevity-large-partitions-200k-pks-db-node-384d3370-10 !INFO    | scylla[787427]:   /opt/scylladb/libreloc/libc.so.6+0x281e1
2021-08-24T05:31:53+00:00  longevity-large-partitions-200k-pks-db-node-384d3370-10 !INFO    | scylla[787427]:   0xe88f2d
```
... 30 minutes later
```
2021-08-24T05:48:03+00:00  longevity-large-partitions-200k-pks-db-node-384d3370-10 !INFO    | scylla:  [shard 0] init - Shutting down database was successful
2021-08-24T05:48:03+00:00  longevity-large-partitions-200k-pks-db-node-384d3370-10 !INFO    | scylla:  [shard 0] init - Shutting down service_memory_limiter
2021-08-24T05:48:03+00:00  longevity-large-partitions-200k-pks-db-node-384d3370-10 !INFO    | scylla:  [shard 0] init - Shutting down service_memory_limiter was successful
2021-08-24T05:48:03+00:00  longevity-large-partitions-200k-pks-db-node-384d3370-10 !INFO    | scylla:  [shard 0] init - Shutting down sst_dir_semaphore
2021-08-24T05:48:03+00:00  longevity-large-partitions-200k-pks-db-node-384d3370-10 !INFO    | scylla:  [shard 0] init - Shutting down sst_dir_semaphore was successful
2021-08-24T05:48:03+00:00  longevity-large-partitions-200k-pks-db-node-384d3370-10 !INFO    | scylla:  [shard 0] init - Shutting down messaging service
2021-08-24T05:48:03+00:00  longevity-large-partitions-200k-pks-db-node-384d3370-10 !INFO    | scylla:  [shard 0] init - Shutting down messaging service was successful
2021-08-24T05:48:03+00:00  longevity-large-partitions-200k-pks-db-node-384d3370-10 !INFO    | scylla:  [shard 0] init - Shutting down API server
2021-08-24T05:48:03+00:00  longevity-large-partitions-200k-pks-db-node-384d3370-10 !INFO    | scylla:  [shard 0] init - Shutting down API server was successful
2021-08-24T05:48:03+00:00  longevity-large-partitions-200k-pks-db-node-384d3370-10 !INFO    | scylla:  [shard 0] init - Shutting down tracing instance
2021-08-24T05:48:03+00:00  longevity-large-partitions-200k-pks-db-node-384d3370-10 !INFO    | scylla:  [shard 0] init - Shutting down tracing instance was successful
2021-08-24T05:48:03+00:00  longevity-large-partitions-200k-pks-db-node-384d3370-10 !INFO    | scylla:  [shard 0] init - Shutting down migration manager notifier
2021-08-24T05:48:03+00:00  longevity-large-partitions-200k-pks-db-node-384d3370-10 !INFO    | scylla:  [shard 0] init - Shutting down migration manager notifier was successful
2021-08-24T05:48:03+00:00  longevity-large-partitions-200k-pks-db-node-384d3370-10 !INFO    | scylla:  [shard 0] init - Shutting down prometheus API server
2021-08-24T05:48:03+00:00  longevity-large-partitions-200k-pks-db-node-384d3370-10 !INFO    | scylla:  [shard 0] init - Shutting down prometheus API server was successful
2021-08-24T05:48:03+00:00  longevity-large-partitions-200k-pks-db-node-384d3370-10 !INFO    | scylla:  [shard 0] init - Shutting down sighup
2021-08-24T05:48:03+00:00  longevity-large-partitions-200k-pks-db-node-384d3370-10 !INFO    | scylla:  [shard 0] init - Shutting down sighup was successful
2021-08-24T05:48:03+00:00  longevity-large-partitions-200k-pks-db-node-384d3370-10 !INFO    | scylla:  [shard 0] init - Scylla version 4.5.rc6-0.20210817.88314fedf shutdown complete.
```

The node is still alive at 52.210.114.55 (scylla service currently DN)

Possibly related issues: https://github.com/scylladb/scylla/issues/8933 https://github.com/scylladb/scylla/issues/9158

Logs:
```
+------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
|                                                                                                                        Log links for testrun with test id 384d3370-a57d-4ac6-a396-13b71f327114                                                                                                                         |
+-----------------+-------------+----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| Date            | Log type    | Link                                                                                                                                                                                                                                                                                   |
+-----------------+-------------+----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| 20210823_111142 | db-cluster  | https://cloudius-jenkins-test.s3.amazonaws.com/384d3370-a57d-4ac6-a396-13b71f327114/20210823_111142/db-cluster-384d3370.tar.gz                                                                                                                                                         |
| 20210823_111142 | loader-set  | https://cloudius-jenkins-test.s3.amazonaws.com/384d3370-a57d-4ac6-a396-13b71f327114/20210823_111142/loader-set-384d3370.tar.gz                                                                                                                                                         |
| 20210823_111142 | monitor-set | https://cloudius-jenkins-test.s3.amazonaws.com/384d3370-a57d-4ac6-a396-13b71f327114/20210823_111142/monitor-set-384d3370.tar.gz                                                                                                                                                        |
| 20210824_054215 | grafana     | https://cloudius-jenkins-test.s3.amazonaws.com/384d3370-a57d-4ac6-a396-13b71f327114/20210824_054215/grafana-screenshot-longevity-large-partition-200k-pks-4days-test-scylla-per-server-metrics-nemesis-20210824_054616-longevity-large-partitions-200k-pks-monitor-node-384d3370-1.png |
| 20210824_055215 | db-cluster  | https://cloudius-jenkins-test.s3.amazonaws.com/384d3370-a57d-4ac6-a396-13b71f327114/20210824_055215/db-cluster-384d3370.tar.gz                                                                                                                                                         |
| 20210824_055215 | loader-set  | https://cloudius-jenkins-test.s3.amazonaws.com/384d3370-a57d-4ac6-a396-13b71f327114/20210824_055215/loader-set-384d3370.tar.gz                                                                                                                                                         |
| 20210824_055215 | monitor-set | https://cloudius-jenkins-test.s3.amazonaws.com/384d3370-a57d-4ac6-a396-13b71f327114/20210824_055215/monitor-set-384d3370.tar.gz                                                                                                                                                        |
| 20210824_055215 | sct-runner  | https://cloudius-jenkins-test.s3.amazonaws.com/384d3370-a57d-4ac6-a396-13b71f327114/20210824_055215/sct-runner-384d3370.tar.gz                                                                                                                                                         |
+-----------------+-------------+----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
```

Monitor: http://34.245.79.215:3000/d/Uc2gPwnnk/longevity-large-partition-200k-pks-4days-test-scylla-per-server-metrics-nemesis-master

Living nodes:
```
  | longevity-large-partitions-200k-pks-db-node-384d3370-1 | 54.216.113.13 | 10.0.0.152 |  
  | longevity-large-partitions-200k-pks-db-node-384d3370-2 | 34.242.187.159 | 10.0.3.55 |  
  | longevity-large-partitions-200k-pks-db-node-384d3370-6 | 54.194.72.108 | 10.0.1.188 |  
  | longevity-large-partitions-200k-pks-db-node-384d3370-7 | 34.240.15.109 | 10.0.1.65 |  
  | longevity-large-partitions-200k-pks-db-node-384d3370-10 | 52.210.114.55 | 10.0.2.112 |  
  | longevity-large-partitions-200k-pks-db-node-384d3370-11 | 18.203.220.255 | 10.0.3.139 |  
```
### Please let me know if I can terminate this cluster

**Comment by slivne** [comment]
Did we get a coredump ?

**Comment by ShlomiBalalis** [comment]
> Did we get a coredump ?

No, there was no coredump

**Comment by roydahan** [comment]
We set the service to "stop-sigterm" so we will have a coredump in these cases, but it doesn't look like it's producing any.

**Comment by ShlomiBalalis** [comment]
I kept node 10 and the cluster alive, the rest of the cluster was terminated. Translated backtrace:
```
void seastar::backtrace<seastar::backtrace_buffer::append_backtrace()::{lambda(seastar::frame)#1}>(seastar::backtrace_buffer::append_backtrace()::{lambda(seastar::frame)#1}&&) at ./build/release/seastar/./seastar/include/seastar/util/backtrace.hh:59
 (inlined by) seastar::backtrace_buffer::append_backtrace() at ./build/release/seastar/./seastar/src/core/reactor.cc:758
 (inlined by) seastar::print_with_backtrace(seastar::backtrace_buffer&, bool) at ./build/release/seastar/./seastar/src/core/reactor.cc:788
seastar::print_with_backtrace(char const*, bool) at ./build/release/seastar/./seastar/src/core/reactor.cc:800
 (inlined by) seastar::sigabrt_action() at ./build/release/seastar/./seastar/src/core/reactor.cc:3597
 (inlined by) operator() at ./build/release/seastar/./seastar/src/core/reactor.cc:3579
 (inlined by) __invoke at ./build/release/seastar/./seastar/src/core/reactor.cc:3575
?? ??:0
?? ??:0
seastar::internal::io_pgetevents(unsigned long, long, long, seastar::internal::linux_abi::io_event*, timespec const*, __sigset_t const*, bool) at ./build/release/seastar/./seastar/src/core/linux-aio.cc:150
seastar::reactor_backend_aio::await_events(int, __sigset_t const*) at ./build/release/seastar/./seastar/src/core/reactor_backend.cc:402
seastar::reactor_backend_aio::wait_and_process_events(__sigset_t const*) at ./build/release/seastar/./seastar/src/core/reactor_backend.cc:469
seastar::reactor::sleep() at ./build/release/seastar/./seastar/src/core/reactor.cc:2916
 (inlined by) seastar::reactor::run() at ./build/release/seastar/./seastar/src/core/reactor.cc:2882
seastar::app_template::run_deprecated(int, char**, std::function<void ()>&&) at ./build/release/seastar/./seastar/src/core/app-template.cc:207
seastar::app_template::run(int, char**, std::function<seastar::future<int> ()>&&) at ./build/release/seastar/./seastar/src/core/app-template.cc:115
main at ./main.cc:493
?? ??:0
_start at ??:?

```

**Comment by slivne** [comment]
@roydahan I guess we need to run it on the side to see it works and fix it so  it will actually produce what we want for this cases.

**Comment by slivne** [comment]
not sure we can do anything with this info - but we are stuck in shutdown  maybe its worth to check that we actually completed startup before shutdown was requested.

**Comment by avikivity** [comment]
I don't understand the logs. At 2021-08-24T05:31:53+00:00 systemd gave up and sent SIGABRT. Scylla immediately acknowledged and crashed. But at 2021-08-24T05:48:03+00:00 the shutdown completed. This is impossible.

Are there no logs in between? The only explanation I see is that scylla restarted and was shut down again.

**Comment by ShlomiBalalis** [comment]
> not sure we can do anything with this info - but we are stuck in shutdown maybe its worth to check that we actually completed startup before shutdown was requested.

Yes, we make sure that scylla was started when we command it to start. And specifically here:
`2021-08-24T05:06:17+00:00  longevity-large-partitions-200k-pks-db-node-384d3370-10 !INFO    | scylla:  [shard 0] init - Scylla version 4.5.rc6-0.20210817.88314fedf initialization completed.`
It was up 10 minutes before the shut down started

> I don't understand the logs. At 2021-08-24T05:31:53+00:00 systemd gave up and sent SIGABRT. Scylla immediately acknowledged and crashed. But at 2021-08-24T05:48:03+00:00 the shutdown completed. This is impossible.
> 
> Are there no logs in between? The only explanation I see is that scylla restarted and was shut down again.

I don't see any log from scylla between those timestamps, but you are correct in a sense, I missed a part of the nemesis. When we execute the `sudo systemctl stop scylla-server.service` command, we have a 5 minutes timeout and 2 retries. So the command was not only executed at `05:16:52,685`, but also at `05:21:57,697` and `05:27:02,708`, which explains the "holes" in the log and the sudden appearance of the shard abortion. However, between `2021-08-24T05:31:53` and `2021-08-24T05:48:03` we did no execute another command.

**Comment by fgelcer** [comment]
i'm seeing such issue, in my case during `nodetool drain`, but the trace seems very similar:

```
2022-06-06 21:08:52.607 <22-06-06 21:08:52.000>: (DatabaseLogEvent Severity.WARNING) period_type=one-time event_id=ce1a3d17-1881-4132-bae3-60b48e481edc: type=RESTARTED_DUE_TO_TIME_OUT regex=scylla-server.service.*State 'stop-sigterm' timed out line_number=103960 node=longev
ity-10gb-3h-master-db-node-23532d5d-0-6
022-06-06T21:08:52+00:00 longevity-10gb-3h-master-db-node-23532d5d-0-6  !WARNING | systemd[1]: scylla-server.service: State 'stop-sigterm' timed out. Killing.
2022-06-06 21:08:52.607 <2022-06-06 21:08:52.000>: (DatabaseLogEvent Severity.ERROR) period_type=one-time event_id=8260b0d6-cf79-4df7-991d-8dbff1d6ffed: type=ABORTING_ON_SHARD regex=Aborting on shard line_number=103962 node=longevity-10gb-3h-master-db-node-23532d5d-0-6
2022-06-06T21:08:52+00:00 longevity-10gb-3h-master-db-node-23532d5d-0-6     !INFO | scylla[42553]: Aborting on shard 0.
void seastar::backtrace<seastar::backtrace_buffer::append_backtrace()::{lambda(seastar::frame)#1}>(seastar::backtrace_buffer::append_backtrace()::{lambda(seastar::frame)#1}&&) at ./build/release/seastar/./seastar/include/seastar/util/backtrace.hh:59
 (inlined by) seastar::backtrace_buffer::append_backtrace() at ./build/release/seastar/./seastar/src/core/reactor.cc:758
 (inlined by) seastar::print_with_backtrace(seastar::backtrace_buffer&, bool) at ./build/release/seastar/./seastar/src/core/reactor.cc:788
seastar::print_with_backtrace(char const*, bool) at ./build/release/seastar/./seastar/src/core/reactor.cc:800
 (inlined by) seastar::sigabrt_action() at ./build/release/seastar/./seastar/src/core/reactor.cc:3683
 (inlined by) operator() at ./build/release/seastar/./seastar/src/core/reactor.cc:3665
 (inlined by) __invoke at ./build/release/seastar/./seastar/src/core/reactor.cc:3661
?? ??:0
?? ??:0
seastar::internal::io_pgetevents(unsigned long, long, long, seastar::internal::linux_abi::io_event*, timespec const*, __sigset_t const*, bool) at ./build/release/seastar/./seastar/src/core/linux-aio.cc:150
seastar::reactor_backend_aio::await_events(int, __sigset_t const*) at ./build/release/seastar/./seastar/src/core/reactor_backend.cc:457
seastar::reactor_backend_aio::wait_and_process_events(__sigset_t const*) at ./build/release/seastar/./seastar/src/core/reactor_backend.cc:524
seastar::reactor::sleep() at ./build/release/seastar/./seastar/src/core/reactor.cc:3024
 (inlined by) seastar::reactor::do_run() at ./build/release/seastar/./seastar/src/core/reactor.cc:2990
seastar::reactor::run() at ./build/release/seastar/./seastar/src/core/reactor.cc:2821
seastar::app_template::run_deprecated(int, char**, std::function<void ()>&&) at ./build/release/seastar/./seastar/src/core/app-template.cc:265
seastar::app_template::run(int, char**, std::function<seastar::future<int> ()>&&) at ./build/release/seastar/./seastar/src/core/app-template.cc:156
scylla_main(int, char**) at ./main.cc:535
std::function<int (int, char**)>::operator()(int, char**) const at /usr/lib/gcc/x86_64-redhat-linux/11/../../../../include/c++/11/bits/std_function.h:590
 (inlined by) main at ./main.cc:1575
?? ??:0
_start at ??:?
```


## Installation details

Kernel Version: 5.13.0-1027-gcp
Scylla version (or git commit hash): `5.1.dev-20220606.605ee74c39b2` with build-id `8d11831b4f5157cf34644edbc6314947acebd2b7`
Cluster size: 6 nodes (n1-highmem-16)

Scylla Nodes used in this run:
- longevity-10gb-3h-master-db-node-23532d5d-0-8 (34.139.25.233 | 10.142.0.67) (shards: 14)
- longevity-10gb-3h-master-db-node-23532d5d-0-7 (34.74.162.189 | 10.142.0.66) (shards: 14)
- longevity-10gb-3h-master-db-node-23532d5d-0-6 (34.139.67.156 | 10.142.0.48) (shards: 14)
- longevity-10gb-3h-master-db-node-23532d5d-0-5 (34.148.241.90 | 10.142.0.47) (shards: 14)
- longevity-10gb-3h-master-db-node-23532d5d-0-4 (34.139.130.254 | 10.142.0.46) (shards: 14)
- longevity-10gb-3h-master-db-node-23532d5d-0-3 (35.243.170.93 | 10.142.0.44) (shards: 14)
- longevity-10gb-3h-master-db-node-23532d5d-0-2 (34.139.189.242 | 10.142.0.43) (shards: 14)
- longevity-10gb-3h-master-db-node-23532d5d-0-1 (34.148.128.27 | 10.142.0.42) (shards: 14)


OS / Image: `https://www.googleapis.com/compute/v1/projects/scylla-images/global/images/217205913884529810` (gce: us-east1)

Test: `longevity-10gb-3h-gce-test`
Test id: `23532d5d-199a-4a29-9ef8-791fad287098`
Test name: `scylla-master/longevity/longevity-10gb-3h-gce-test`
Test config file(s):

- [longevity-10gb-3h.yaml](https://github.com/scylladb/scylla-cluster-tests/blob/5c1a8968e7a2c1bbbca61fa40c6da380a40cd939/test-cases/longevity/longevity-10gb-3h.yaml)


- Restore Monitor Stack command: `$ hydra investigate show-monitor 23532d5d-199a-4a29-9ef8-791fad287098`
- Restore monitor on AWS instance using [Jenkins job](https://jenkins.scylladb.com/view/QA/job/QA-tools/job/hydra-show-monitor/parambuild/?test_id=23532d5d-199a-4a29-9ef8-791fad287098)
- Show all stored logs command: `$ hydra investigate show-logs 23532d5d-199a-4a29-9ef8-791fad287098`

## Logs:
- **db-cluster-23532d5d.tar.gz** - [https://cloudius-jenkins-test.s3.amazonaws.com/23532d5d-199a-4a29-9ef8-791fad287098/20220606_220920/db-cluster-23532d5d.tar.gz](https://cloudius-jenkins-test.s3.amazonaws.com/23532d5d-199a-4a29-9ef8-791fad287098/20220606_220920/db-cluster-23532d5d.tar.gz)
- **monitor-set-23532d5d.tar.gz** - [https://cloudius-jenkins-test.s3.amazonaws.com/23532d5d-199a-4a29-9ef8-791fad287098/20220606_220920/monitor-set-23532d5d.tar.gz](https://cloudius-jenkins-test.s3.amazonaws.com/23532d5d-199a-4a29-9ef8-791fad287098/20220606_220920/monitor-set-23532d5d.tar.gz)
- **loader-set-23532d5d.tar.gz** - [https://cloudius-jenkins-test.s3.amazonaws.com/23532d5d-199a-4a29-9ef8-791fad287098/20220606_220920/loader-set-23532d5d.tar.gz](https://cloudius-jenkins-test.s3.amazonaws.com/23532d5d-199a-4a29-9ef8-791fad287098/20220606_220920/loader-set-23532d5d.tar.gz)
- **sct-runner-23532d5d.tar.gz** - [https://cloudius-jenkins-test.s3.amazonaws.com/23532d5d-199a-4a29-9ef8-791fad287098/20220606_220920/sct-runner-23532d5d.tar.gz](https://cloudius-jenkins-test.s3.amazonaws.com/23532d5d-199a-4a29-9ef8-791fad287098/20220606_220920/sct-runner-23532d5d.tar.gz)
- **parallel-timelines-report-23532d5d.tar.gz** - [https://cloudius-jenkins-test.s3.amazonaws.com/23532d5d-199a-4a29-9ef8-791fad287098/20220606_220920/parallel-timelines-report-23532d5d.tar.gz](https://cloudius-jenkins-test.s3.amazonaws.com/23532d5d-199a-4a29-9ef8-791fad287098/20220606_220920/parallel-timelines-report-23532d5d.tar.gz)


[Jenkins job URL](https://jenkins.scylladb.com/job/scylla-master/job/longevity/job/longevity-10gb-3h-gce-test/183/)

**Comment by avikivity** [comment]
```
Jun 06 20:53:54 longevity-10gb-3h-master-db-node-23532d5d-0-6 scylla[42553]:  [shard 0] init - Shutting down forward service was successful
Jun 06 20:53:54 longevity-10gb-3h-master-db-node-23532d5d-0-6 scylla[42553]:  [shard 0] init - Shutting down database
Jun 06 21:08:52 longevity-10gb-3h-master-db-node-23532d5d-0-6 systemd[1]: scylla-server.service: State 'stop-sigterm' timed out. Killing.
Jun 06 21:08:52 longevity-10gb-3h-master-db-node-23532d5d-0-6 systemd[1]: scylla-server.service: Killing process 42553 (scylla) with signal SIGABRT.
Jun 06 21:08:52 longevity-10gb-3h-master-db-node-23532d5d-0-6 scylla[42553]: Aborting on shard 0.
Jun 06 21:08:52 longevity-10gb-3h-master-db-node-23532d5d-0-6 scylla[42553]: Backtrace:
Jun 06 21:08:52 longevity-10gb-3h-master-db-node-23532d5d-0-6 scylla[42553]:   0x4d66298
Jun 06 21:08:52 longevity-10gb-3h-master-db-node-23532d5d-0-6 scylla[42553]:   0x4d97a82
Jun 06 21:08:52 longevity-10gb-3h-master-db-node-23532d5d-0-6 scylla[42553]:   0x7fab20c14a1f
Jun 06 21:08:52 longevity-10gb-3h-master-db-node-23532d5d-0-6 scylla[42553]:   /opt/scylladb/libreloc/libc.so.6+0xfaddc
Jun 06 21:08:52 longevity-10gb-3h-master-db-node-23532d5d-0-6 scylla[42553]:   0x4dc0310
Jun 06 21:08:52 longevity-10gb-3h-master-db-node-23532d5d-0-6 scylla[42553]:   0x4dbbf04
Jun 06 21:08:52 longevity-10gb-3h-master-db-node-23532d5d-0-6 scylla[42553]:   0x4dbc673
Jun 06 21:08:52 longevity-10gb-3h-master-db-node-23532d5d-0-6 scylla[42553]:   0x4d7894c
Jun 06 21:08:52 longevity-10gb-3h-master-db-node-23532d5d-0-6 scylla[42553]:   0x4d7781c
Jun 06 21:08:52 longevity-10gb-3h-master-db-node-23532d5d-0-6 scylla[42553]:   0x4d1e858
Jun 06 21:08:52 longevity-10gb-3h-master-db-node-23532d5d-0-6 scylla[42553]:   0x4d1dd31
Jun 06 21:08:52 longevity-10gb-3h-master-db-node-23532d5d-0-6 scylla[42553]:   0x105afef
Jun 06 21:08:52 longevity-10gb-3h-master-db-node-23532d5d-0-6 scylla[42553]:   0x105881a
Jun 06 21:08:52 longevity-10gb-3h-master-db-node-23532d5d-0-6 scylla[42553]:   /opt/scylladb/libreloc/libc.so.6+0x27b74
Jun 06 21:08:52 longevity-10gb-3h-master-db-node-23532d5d-0-6 scylla[42553]:   0x10577ad
Jun 06 21:23:52 longevity-10gb-3h-master-db-node-23532d5d-0-6 systemd[1]: scylla-server.service: Processes still around after SIGKILL. Ignoring.
Jun 06 21:38:52 longevity-10gb-3h-master-db-node-23532d5d-0-6 systemd[1]: scylla-server.service: State 'stop-post' timed out. Terminating.
Jun 06 21:53:52 longevity-10gb-3h-master-db-node-23532d5d-0-6 systemd[1]: scylla-server.service: State 'stop-final-sigterm' timed out. Killing.
Jun 06 21:53:52 longevity-10gb-3h-master-db-node-23532d5d-0-6 systemd[1]: scylla-server.service: Killing process 42553 (scylla) with signal SIGABRT.
Jun 06 21:57:24 longevity-10gb-3h-master-db-node-23532d5d-0-6 systemd[1]: scylla-server.service: Main process exited, code=dumped, status=6/ABRT
Jun 06 21:57:24 longevity-10gb-3h-master-db-node-23532d5d-0-6 systemd[1]: scylla-server.service: Failed with result 'timeout'.
Jun 06 21:57:24 longevity-10gb-3h-master-db-node-23532d5d-0-6 systemd[1]: Starting Scylla Server...
Jun 06 21:57:24 longevity-10gb-3h-master-db-node-23532d5d-0-6 scylla_prepare[59741]: irqbalance is not running
Jun 06 21:57:24 longevity-10gb-3h-master-db-node-23532d5d-0-6 scylla_prepare[59741]: No non-NVMe disks to tune
```

"Shutting down database" was slow, so we generated a core dump with the abort signal. Where is the core dump?

**Comment by avikivity** [comment]
Maybe the core dump generation wasn't successful.

**Comment by mykaul** [comment]
Based on https://github.com/scylladb/scylla-enterprise/issues/2790#issuecomment-1478021541 this is fixed in 5.1, but needs backport (to 5.0 and then to 2021.1). Closing and adding backport-candidate label.

**Comment by denesb** [comment]
Backported to 5.0 and 2021.1
